### PR TITLE
add the support of the ADC, watchdog and entropy for the stm32h7rs 

### PR DESCRIPTION
**Pull request by FRASTM**
add the ADC peripheral  ADC1 & 2 instances
add the watchdog IWDG and WWDG peripherals
add the RNG peripheral for entropy generator
to the stm32H7R/S serie 

Enables the nodes on the stm32h7s78_dk disco kit

### CODE REVIEW

**Comment by FRASTM** [comment]
Running the   tests/drivers/entropy/api/  on the stm32h7s78_dk
```
*** Booting Zephyr OS build v3.6.0-5496-gb2c5b85078cb ***
Running TESTSUITE entropy_api
===================================================================
START - test_entropy_get_entropy
random device is 0x800749c, name is rng@48020000
  0x51
  0x2c
  0x63
  0x98
  0xb0
  0x7d
  0xdb
  0xe9                                                                          
  0x28                                                                          
 PASS - test_entropy_get_entropy in 0.011 seconds                               
===================================================================             
TESTSUITE entropy_api succeeded                                                 
                                                                                
------ TESTSUITE SUMMARY START ------                                           
                                                                                
SUITE PASS - 100.00% [entropy_api]: pass = 1, fail = 0, skip = 0, total = 1 durs
 - PASS - [entropy_api.test_entropy_get_entropy] duration = 0.011 seconds       
                                                                                
------ TESTSUITE SUMMARY END ------                                             
                                                                                
===================================================================             
PROJECT EXECUTION SUCCESSFUL   
```

**Comment by FRASTM** [comment]
running the samples/sensor/die_temp_polling/ on the stm32h7s78_dk:
```
*** Booting Zephyr OS build v3.6.0-5496-gb2c5b85078cb ***               
CPU Die temperature[dietemp]: 25.3 �°C                                          
CPU Die temperature[dietemp]: 25.7 �°C  
```

**Comment by FRASTM** [comment]
running the samples/subsys/task_wdt/

```
*** Booting Zephyr OS build v3.6.0-5496-gb2c5b85078cb ***
Task watchdog sample application.
Main thread still alive...
Control thread started.
Main thread still alive...
Main thread still alive...
Main thread still alive...
Control thread getting stuck...
Task watchdog channel 1 callback, thread: control
Resetting device...
*** Booting Zephyr OS build v3.6.0-5496-gb2c5b85078cb ***
Task watchdog sample application.
...
```

**Review by erwango — DISMISSED** [review]

File: boards/st/stm32h7s78_dk/stm32h7s78.yaml [hunk 1]
```diff
@@ -10,4 +10,11 @@ supported:
   - arduino_gpio
   - gpio
   - uart
+  - watchdog
+  - entropy
+  - adc
+  - dac
```
Comment (ajarmouni-st) [inline boards/st/stm32h7s78_dk/stm32h7s78.yaml:16]: DAC is not enabled in dts

Comment (FRASTM) [reply boards/st/stm32h7s78_dk/stm32h7s78.yaml:16]: correct, removed

**Review by ajarmouni-st — DISMISSED** [review]

File: drivers/adc/adc_stm32.c [hunk 2]
```diff
@@ -217,7 +217,7 @@ static void adc_stm32_enable_dma_support(ADC_TypeDef *adc)
 {
 	/* Allow ADC to create DMA request and set to one-shot mode as implemented in HAL drivers */
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X)
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H7RSX)
```
Comment (gautierg-st) [inline drivers/adc/adc_stm32.c:220]: Are you sure about that? I don't find the function LL_ADC_REG_SetDataTransferMode or LL_ADC_REG_SetDMATransferMode in the LL H7RS.
But there is the LL_ADC_REG_SetDMATransfer function, so I think that for H7RS we should fall into the `#else` case.

**Review by gautierg-st — CHANGES_REQUESTED** [review]
I think a change is needed for the DMA part of the ADC driver

File: drivers/adc/adc_stm32.c (hunk 2 above)
Comment (FRASTM) [reply drivers/adc/adc_stm32.c]: correct, this is not valid for the ADC of the stm32h7RS

**Review by gautierg-st — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by ajarmouni-st — DISMISSED** [review]

**Comment by FRASTM** [comment]
CI failure is fixed by https://github.com/zephyrproject-rtos/zephyr/pull/74075/

**Comment by FRASTM** [comment]
Rebase on df788d0b41300473f047a1497fae57e8ab351f26 with https://github.com/zephyrproject-rtos/zephyr/pull/74075
fix CI error with flash driver updated to support the stm32h7rs

**Review by ajarmouni-st — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**